pkg/informer/informer.go

```diff
@@ -68,6 +70,11 @@ type DynamicDiscoverySharedInformerFactory struct {
 	informers        map[schema.GroupVersionResource]informers.GenericInformer
 	startedInformers map[schema.GroupVersionResource]bool
 	informerStops    map[schema.GroupVersionResource]chan struct{}
+	discoveryData    []*metav1.APIResourceList
+
+	// Support subscribers (e.g. quota) that want to know when informers/discovery have changed.
+	subscribersLock sync.Mutex
+	subscribers     map[string]chan<- struct{}
```

Comment:
my comment still stands. This is a bad interface. Use a function call please and document that these must return instantly.